stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -613,9 +613,9 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
         }
     }
     
-    public subscript(bounds: Range<Index>) -> MutableRandomAccessSlice<Data> {
+    public subscript(bounds: Range<Index>) -> MutableRangeReplaceableRandomAccessSlice<Data> {
         get {
-            return MutableRandomAccessSlice(base: self, bounds: bounds)
+            return MutableRangeReplaceableRandomAccessSlice(base: self, bounds: bounds)
```

Comment:
Alternatively, you can use `Data` as `Data`'s slice type (I was just making a minimal change).  It would make sense if constructing a `Data` pointing to other `Data`'s storage is O(1) (I don't know if it is).